src/views/ThreeWProjectForm/index.tsx

```diff
@@ -0,0 +1,1004 @@
+    } = useContext(RouteContext);
+
+    const {
+        value,
+        setValue,
+        setFieldValue,
+        error: formError,
+        setError: onErrorSet,
+        validate,
+    } = useForm(schema, { value: defaultFormValues });
+
+    const navigate = useNavigate();
+
+    const [districtOptions, setDistrictOptions] = useState<
+        DistrictItem[] | undefined | null
+    >([]);
+
+    const [events, setEvents] = useState<
+        EventItem[] | undefined | null
+    >([]);
+
+    const { pending: pendingProjectDetails } = useRequest({
+        skip: isFalsyString(projectId),
+        url: '/api/v2/project/{id}/',
+        pathVariables: isTruthyString(projectId) ? {
+            id: Number(projectId),
+        } : undefined,
+        onSuccess: (response) => {
+            setDistrictOptions(response.project_districts_detail);
+            setEvents([{
+                ...response.event_detail,
+                // FIXME: event dtype id is a must inside event mini but
+                // its not defined under event_detail of this response
+                dtype: { id: response.event_detail?.dtype } as EventItem['dtype'],
+            }]);
+            setValue({
+                ...response,
+                // Set beginning is_annual_report switch according to
+                // the filled-in annual split details
+                is_annual_report: response.annual_split_detail?.length >= 1,
+                annual_split_detail: response.annual_split_detail.map((split) => ({
+                    ...injectClientId(split),
+                })),
+            });
+        },
+    });
+
+    const {
+        deployments_project_status: projectStatusOptions,
+        deployments_project_operation_type: operationTypeOptions,
+        deployments_project_programme_type: programmeTypeOptions,
+        api_visibility_char_choices: visibilityTypeOptions,
+    } = useContext(ServerEnumsContext);
+
+    const projectStatusOptionsMap = listToMap(
+        projectStatusOptions,
+        (d) => d.key,
+        (d) => d.value,
+    );
+
+    const error = getErrorObject(formError);
+    const handleProjectCountryChange = useCallback(
+        (val: number | undefined, name: 'project_country') => {
+            setFieldValue(val, name);
+            setFieldValue(undefined, 'project_districts' as const);
+            // FIXME: Add admin2 select input
+        },
+        [setFieldValue],
+    );
+
+    const isTotalRequired = value.status === PROJECT_STATUS_COMPLETED;
+
+    const shouldDisableTotalTarget = !isFalsy(value.target_male)
+        || !isFalsy(value.target_female)
+        || !isFalsy(value.target_other);
+
+    const shouldDisableTotalReached = !isFalsy(value.reached_male)
+        || !isFalsy(value.reached_female)
+        || !isFalsy(value.reached_other);
+
+    const {
+        pending: fetchingPrimarySectors,
+        response: primarySectorOptions,
+    } = useRequest({
+        url: '/api/v2/primarysector',
+    });
```

Comment:
I was thinking if we can create lazy select input for these kind of select inputs as well

Reply:
These are fixed in the server and are limited to 12 or 10, so added this

Reply:
We also have a new proposal that can improve performance for these requests